sample-app/src/components/DropdownSection.tsx

```diff
@@ -0,0 +1,108 @@
+import classNames from "classnames";
+import { useState, useEffect } from "react";
+
+export interface Option {
+  value: string,
+  display: JSX.Element
+}
+
+interface DropdownSectionProps<T> {
+  focusStatus?: string,
+  options: T[],
+  optionIdPrefix: string,
+  onFocusChange: (value: string, focusedOptionId: string) => void,
+  onLeaveSectionFocus?: (focusNext: boolean) => void,
+  onClickOption: (option: T, optionIndex: number) => void,
+  label?: string,
+  cssClasses: {
+    sectionContainer: string,
+    sectionLabel: string,
+    optionsContainer: string,
+    option: string,
+    focusedOption: string
+  }
+}
+
+export default function DropdownSection<T extends Option>({
+  focusStatus,
+  options,
+  optionIdPrefix,
+  onFocusChange,
+  onLeaveSectionFocus = () => {},
+  onClickOption,
+  label = '',
+  cssClasses
+}: DropdownSectionProps<T>): JSX.Element | null {
+
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState<number>(0);
```

Comment:
should the initial state be -1 instead? since section is not focused by default.

Reply:
right now the logic works by always pointing to the first valid option index (which would be 0). that way, as soon as focus shifts to it, the index will be valid. `focusStatus` is indicating whether the section is in focus or not

Reply:
`focusedSectionIndex` is undefined by default, in a similar sense I would expect this to be -1 by default as well? From my understanding, all section would be mark as 'inactive' first before setting to 'active', so it can set the index to 0 then (this would help get rid of reset - from my other comment in the pr).

Reply:
It would work either way. I could make it -1 or undefined and then change it to 0 when the section becomes active. In InputDropdown, it makes sense to start with undefined because it has knowledge of the input bar that the section doesn't have. So, I think I slightly prefer it this way so that the sections always point to a valid option index and don't have the notion of pointing to something before their first option.